### PR TITLE
Add title attributes to all items in the repo list viewer

### LINKED ISSUE #6256: Long filenames are indistinguishable if they are shortened to common substring.
File names on directory listing are shortened with "..." if they are too long. When there are multiple files with long name that differ at the very end, they may turn out to be all shortened to the same name making it hard for user to tell them apart.

- Gitea version (or commit ref): ad86b84
- Git version: not relevant
- Operating system: not relevant
- Database (use `[x]`): not relevant
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes ( https://try.gitea.io/mateusza/..qwe/src/branch/master/4.0 )
  - [ ] No
  - [ ] Not relevant
- Log gist:

## Description

This could be solved by adding title= attribute with full name, so it appears on mouse hover. That's the way GitHub does it.

## Screenshots
image

GitHub:
image

### PR DESCRIPTION
**Pull request by jolheiser**
Resolves #6256

Leaving as a draft for now, going to see if there are more places this should be done as well.

### CODE REVIEW

**Comment by mateusza** [comment]
I noticed that github put full (multiline) commit message into title="" on the same screen... But I'm not sure if this time we want to have this one too... Might be a little waste of bandwith.

**Comment by jolheiser** [comment]
Counting this as ready for review. More enhancements could be made down the road if wanted/needed.

File: templates/repo/view_list.tmpl [hunk 1]
```diff
@@ -65,7 +65,7 @@
 							{{$subJumpablePathName := $entry.GetSubJumpablePathName}}
 							{{$subJumpablePath := SubJumpablePath $subJumpablePathName}}
 							<span class="octicon octicon-file-directory"></span>
-							<a href="{{EscapePound $.TreeLink}}/{{EscapePound $subJumpablePathName}}">
+							<a href="{{EscapePound $.TreeLink}}/{{EscapePound $subJumpablePathName}}" title="{{$subJumpablePathName}}">
```
Comment (lunny) [inline templates/repo/view_list.tmpl:68]: maybe still `EscapePound` ?

**Review by lunny — CHANGES_REQUESTED** [review]

File: templates/repo/view_list.tmpl (hunk 1 above)
Comment (jolheiser) [reply templates/repo/view_list.tmpl:68]: Since this is the title, I don't think it needs to be escaped since `EscapePound` is for URLs , right?

Comment (zeripath) [reply templates/repo/view_list.tmpl:68]: The template library should automatically ensure that the `{{$subJumpablePathName}}` is appropriately encoded for attributes.

**Review by lunny — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

**Comment by zeripath** [comment]
I wonder if this should be backported to 1.8?

**Pull request merged**